src/EnergyPlus/CurveManager.cc

```diff
@@ -2038,9 +2040,9 @@ namespace CurveManager {
 			TableLookup( TableNum ).NumIndependentVars = 1;
 			TableLookup( TableNum ).NumX1Vars = size( PerfCurveTableData( TableNum ).X1 );
 			TableLookup( TableNum ).X1Var.allocate( TableLookup( TableNum ).NumX1Vars );
-			TableLookup( TableNum ).TableLookupZData.allocate( 1, 1, 1, 1, size( PerfCurveTableData( TableNum ).Y ) );
+			TableLookup( TableNum ).TableLookupZData.allocate( 1, 1, 1, 1, 1, size( PerfCurveTableData( TableNum ).Y ) );
```

Comment:
ZData now needs an extra dimension, so wherever it's not used (5 dimensions or less), just include scalar dimensionality like already done for lower rank matrices.